src/Client.php

```diff
@@ -81,6 +81,13 @@ public function connect()
         $this->socket->connect($endpoint);
     }
 
+    public function reconnect()
+    {
+        $endpoint = $this->context->hookResolveEndpoint($this->_endpoint, $this->_version);
+        $this->socket->disconnect($endpoint);
+        $this->socket->connect($endpoint);
```

Comment:
I think we could take 2 ways to handle with it:
1) after timeout, destroy the connection obj. and throw an exception out. end the php script.
2) after timeout, destroy the connection obj. 
     2.1) if we have another request in the following script, we establish the connection again (lazy connection)
     2.2) if we don't have other demand to request data from server, just let it go.

Reply:
Updated, please check.